merkledag/merkledag.go

```diff
@@ -460,16 +443,25 @@ func EnumerateChildrenAsync(ctx context.Context, ds DAGService, c *cid.Cid, visi
 			if inProgress == 0 && next == nil {
 				return nil
 			}
-		case nc := <-out:
-			if nc.Err != nil {
-				return nc.Err
+		case l := <-out:
+			n, err := ds.Get(ctx, l)
+			if err != nil {
+				return err
 			}
 
-			if next == nil {
-				next = nc.Node
-				send = feed
-			} else {
-				todobuffer = append(todobuffer, nc.Node)
+			links := n.Links()
+			if len(links) > 0 {
+				cids := make([]*cid.Cid, len(links), len(links))
```

Comment:
Actually, you can rewrite this whole loop to avoid allocating an extra slice. Just append to the todobuffer